application/src/main/java/org/opentripplanner/graph_builder/module/DirectTransferGenerator.java

```diff
@@ -172,6 +209,21 @@ public void buildGraph() {
       nTransfersTotal,
       nLinkedStops
     );
+    for (StreetMode mode : transferRequests
+      .stream()
+      .map(transferProfile -> transferProfile.journey().transfer().mode())
+      .distinct()
+      .toList()) {
+      LOG.info(
+        "Created {} transfers for mode {}.",
+        transfersByStop
+          .values()
+          .stream()
+          .filter(pathTransfer -> pathTransfer.getModes().contains(mode))
+          .count(),
+        mode
+      );
```

Comment:
This is quite difficult to read. I would probably continue the stream instead of making it a list and then looping over it, and then call some method with the mode as an argument that does the logging. So something like:
```
transferRequests
      .stream()
      .map(transferProfile -> transferProfile.journey().transfer().mode())
      .distinct()
      .peek(mode -> logTransferForMode(mode));
```

Reply:
I changed it to this:
```
transferRequests
      .stream()
      .map(transferProfile -> transferProfile.journey().transfer().mode())
      .distinct()
      .forEach(mode ->
        LOG.info(
          "Created {} transfers for mode {}.",
          timetableRepository.getTransfersByMode(mode).size(),
          mode
        )
      );
```